### PR TITLE
[Release test] Lower bound `botocore` version for AWS cluster launcher test

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
When running the AWS cluster launcher release test, the `botocore` version is outdated, causing an ImportError:

```
(base) ray@ip-10-0-3-28:~/oss-release-tests-2$ pip list | grep boto
boto3                                  1.26.103
botocore                               1.27.6
(base) ray@ip-10-0-3-28:~/oss-release-tests-2$ python -c "import boto3"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/__init__.py", line 17, in <module>
    from boto3.session import Session
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/session.py", line 25, in <module>
    from .resources.factory import ResourceFactory
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/resources/factory.py", line 17, in <module>
    from ..docs import docstring
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/docs/__init__.py", line 15, in <module>
    from botocore.docs import DEPRECATED_SERVICE_NAMES
ImportError: cannot import name 'DEPRECATED_SERVICE_NAMES' from 'botocore.docs' (/home/ray/anaconda3/lib/python3.7/site-packages/botocore/docs/__init__.py)
(base) ray@ip-10-0-3-28:~/oss-release-tests-2$ 
```

Since `botocore` is not packaged with Ray, we should specify it in the `pip` package dependencies for this test.  This PR does that, lower bounding `botocore` at `1.29` so that this import will succeed.

Side note: I spent some time investigating where the old version of `botocore` on the release test cluster came from, but I couldn't figure it out.

## Related issue number
Closes https://github.com/anyscale/product/issues/19019

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
Tests running here https://buildkite.com/ray-project/release-tests-pr/builds/33705

File: python/ray/autoscaler/aws/tests/aws_config.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ base_image: {{ env["RAY_IMAGE_NIGHTLY_CPU"] | default("anyscale/ray:nightly-py37
 debian_packages: []
 
 python:
-  pip_packages: []
+  pip_packages: ["botocore>=1.29"]
```
Comment (wuisawesome) [inline python/ray/autoscaler/aws/tests/aws_config.yaml:5]: Does this mean `ray[default]` is broken right now? If so, can we change it there?

Comment (architkulkarni) [reply python/ray/autoscaler/aws/tests/aws_config.yaml:5]: No, I didn't understand the root cause and could never reproduce the error from scratch.  `ray[default]` in a blank conda environment is fine, and `ray[default]` on top of an Anyscale ray nightly image is also fine

**Comment by architkulkarni** [comment]
This fix didn't work, same error: 
https://buildkite.com/ray-project/release-tests-pr/builds/33705#01874990-8634-42c6-b278-3c37a5f42b79/695-707

**Comment by architkulkarni** [comment]
Weird. Despite this being listed as the pip packages after the cluster environment build is complete:
```
boto3==1.26.105
botocore==1.29.105
```

when I run `pip list` at the beginning of the test script, I get
```
boto3==1.26.105
botocore==1.27.6
```
Link: https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_jkzr6r2pjnb1g9jns3u2juw55a?command-history-section=command_history

**Comment by architkulkarni** [comment]
Ah this is probably the culprit, we're reinstalling `awscli` here: https://buildkite.com/ray-project/release-tests-pr/builds/33781#01874cf8-8426-4ddc-b7c7-b066c17197ba/714-715

`Executing pip install -q awscli && aws s3 cp s3://ray-release-automation-results/tmp/mcfbvhkmix archive.tar.gz && tar xf archive.tar.gz  with {} via ray job submit`

Since it's `pip install awscli` and not `pip install -U awscli`, I believe it's reinstalling the pinned `awscli` version `1.25.6`, which in turn installs `botocore==1.27.6`.

**Comment by architkulkarni** [comment]
Yup, here's a self-contained reproduction:
```
conda create -n boto python=3.8 -y
conda activate boto      
pip install boto3 
pip install awscli==1.25.6
python -c "import boto3"
# Traceback (most recent call last):
#   File "<string>", line 1, in <module>
#   File "/Users/archit/anaconda3/envs/boto/lib/python3.8/site-packages/boto3/__init__.py", line 17, in <module>
#     from boto3.session import Session
#   File "/Users/archit/anaconda3/envs/boto/lib/python3.8/site-packages/boto3/session.py", line 25, in <module>
#     from .resources.factory import ResourceFactory
#   File "/Users/archit/anaconda3/envs/boto/lib/python3.8/site-packages/boto3/resources/factory.py", line 17, in <module>
#     from ..docs import docstring
#   File "/Users/archit/anaconda3/envs/boto/lib/python3.8/site-packages/boto3/docs/__init__.py", line 15, in <module>
#     from botocore.docs import DEPRECATED_SERVICE_NAMES
# ImportError: cannot import name 'DEPRECATED_SERVICE_NAMES' from 'botocore.docs' (/Users/archit/anaconda3/envs/boto/lib/python3.8/site-packages/botocore/docs/__init__.py)
```
I think the right place to fix this is by unpinning `awscli==1.25.6` on the cluster side.  Replacing it with `awscli==1.27.105` in the above reproduction script makes it pass.

**Comment by architkulkarni** [comment]
Closing in favor of https://github.com/anyscale/product/pull/19270 for now

**Pull request closed without merging**